### PR TITLE
Support content rearchitecture

### PR DESCRIPTION
**Pull request by alflennik**
Should produce a APG website with equivalent content and functionality but which supports APG's completely brand-new organization structure.

### CODE REVIEW

**Comment by SteveALee** [comment]
@alflennik the Netlify Preview deploys broke due to a  change in dependencies. I fixed for this branch and also main but it's still showing as broken above (despite my commit to the branch). Hopefully will resolve eventually

Latest preview is https://deploy-preview-170--wai-aria-practices2.netlify.app/aria/apg/

**Comment by alflennik** [comment]
@SteveALee Thank you for fixing the dependency issue, it's great to see that you were able to get the new version of the site to build. Netlify is still not automatically building due to the error `Error fetching branch: https://github.com/w3c/wai-aria-practices pull/170/head`. I did some [research on that error](https://answers.netlify.com/t/error-fetching-branch-from-github-but-it-used-to-work/10941/6) and someone said "all I had to do was clear cache and try again." Would you be willing to try clearing Netlify's cache and see if that fixes the build?

**Comment by SteveALee** [comment]
Sure, but will not be until later tomorrow am UK time.

**Comment by mcking65** [comment]
@alflennik is this PR the final blocking change that will unlock preview of #2417 and enable us to start the review process for the restructure?

I'm a little concerned that we seem stuck here.

@SteveALee what's status on Netlify cache?

**Comment by SteveALee** [comment]
@mcking65 the preview built successfully. Though I did not test it, that should fix it for your next commit  to the PR branch.

You might have sufficient access to manually trigger a build without cache. Netlify adds a bunch of links to the PR.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -24,22 +24,20 @@ jobs:
         ruby-version: 2.6.2
         bundler-cache: true
 
-    - name: Update git submodules
-      run: |
-        git submodule update --recursive --remote
+    # - name: Update git submodules
```
Comment (howard-e) [inline .github/workflows/deploy.yml:27]: Was this commented out just for this PR to work in the current state, as the expected changes aren't available currently on `aria-practices`'s `main` branch? Or they will be removed in the future?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -24,22 +24,20 @@ jobs:
...
-        respec2html --src ./_external/aria-practices/aria-practices.html --out ./_html/aria-practices.html
         npm install -C scripts/pre-build
         node ./scripts/pre-build
 
     - name: Commit changed files and submodule updates
       uses: stefanzweifel/git-auto-commit-action@v4
       with:
         commit_message: Commit changes
-        branch: main
+        # branch: main
```
Comment (howard-e) [inline .github/workflows/deploy.yml:41]: Same question here.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -24,22 +24,20 @@ jobs:
...
-      run: |
-        git submodule update --recursive --remote
+    # - name: Update git submodules
+    #   run: |
+    #     git submodule update --recursive --remote
 
     - name: Update site files
       run: |
         bundle install
-        npm install -g respec
```
Comment (howard-e) [inline .github/workflows/deploy.yml:34]: We discussed making this change obvious in the README.md as it's no longer needed.

**Review by howard-e — APPROVED** [review]
Looks good to me!

Only remaining thing is updating the `README.md` to reflect no longer needing the `respec2html` related commands during the build steps.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (alflennik) [reply .github/workflows/deploy.yml]: Thanks for catching this Howard, this was commented out as part of the debugging for the broken builds. It is not reenabled.

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (alflennik) [reply .github/workflows/deploy.yml:34]: Thanks! Done.

**Comment by alflennik** [comment]
@SteveALee It seems like you are able to manually trigger the builds (?), but unfortunately it seems like when I push new commits, the CI continues to fail and the changes do not show up on the site preview. If you would be willing to share the Netlify credentials I would be willing to spend some time trying to fix it. Or if I can help in some other way, let me know.

@mcking65 We cannot share the link yet with the testers, because the build preview is currently a few commits out of date.

**Comment by SteveALee** [comment]
@alflennik The error is "Error fetching branch: https://github.com/w3c/wai-aria-practices pull/170/head" which makes little sense (I do recall HEAD can get left in a odd state for PRs)

I ran the deploy again without clearing cache and that made no difference. Running with cleared cache works. Then running again without clearing cached fails.

It could be a Netlify bug so I suggest you raise it on their forums. At a minimum we want more visibility on what the error is.

I next work on Thurs day so can look. If you only want a working preview I suggest you use a previous deploy.

**Comment by SteveALee** [comment]
@if that manually trigger deploy link is out of date with your latest commit on the PR branch then the PR HEAD is wrong. That should be fixable. Maybe a commit to master will give the option in this PR that it is out of date and you can then update it

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch

**Comment by alflennik** [comment]
@SteveALee The issue seems to be with Netlify and not this branch, so I am going to merge this and address the Netlify issue in a dedicated PR.

**Pull request merged**